File: go/test/protocol/test_case_action_receive_request.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type testCaseActionReceiveRequest struct {
	Topic            *string           `yaml:"topic"`
	Payload          *string           `yaml:"payload"`
	ContentType      *string           `yaml:"content-type"`
	FormatIndicator  *int              `yaml:"format-indicator"`
	Metadata         map[string]string `yaml:"metadata"`
	CorrelationIndex *int              `yaml:"correlation-index"`
	CorrelationID    *string           `yaml:"correlation-id"`
	Qos              *int              `yaml:"qos"`
	MessageExpiry    *TestCaseDuration `yaml:"message-expiry"`
	ResponseTopic    *string           `yaml:"response-topic"`
	SourceIndex      *int              `yaml:"source-index"`
	PacketIndex      *int              `yaml:"packet-index"`
}

type TestCaseActionReceiveRequest struct {
	testCaseActionReceiveRequest
}

func (receiveRequest *TestCaseActionReceiveRequest) UnmarshalYAML(
	node *yaml.Node,
) error {
	*receiveRequest = TestCaseActionReceiveRequest{}

	receiveRequest.Topic = TestCaseDefaultInfo.Actions.ReceiveRequest.GetTopic()
	receiveRequest.Payload = TestCaseDefaultInfo.Actions.ReceiveRequest.GetPayload()
	receiveRequest.ContentType = TestCaseDefaultInfo.Actions.ReceiveRequest.GetContentType()
	receiveRequest.FormatIndicator = TestCaseDefaultInfo.Actions.ReceiveRequest.GetFormatIndicator()
	receiveRequest.CorrelationIndex = TestCaseDefaultInfo.Actions.ReceiveRequest.GetCorrelationIndex()
	receiveRequest.Qos = TestCaseDefaultInfo.Actions.ReceiveRequest.GetQos()
	receiveRequest.ResponseTopic = TestCaseDefaultInfo.Actions.ReceiveRequest.GetResponseTopic()
	receiveRequest.SourceIndex = TestCaseDefaultInfo.Actions.ReceiveRequest.GetSourceIndex()

	defaultMessageExpiry := TestCaseDefaultInfo.Actions.ReceiveRequest.GetMessageExpiry()
	receiveRequest.MessageExpiry = &defaultMessageExpiry

	return node.Decode(&receiveRequest.testCaseActionReceiveRequest)
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseActionReceiveRequest.cs
namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseActionReceiveRequest : TestCaseAction
    {
        public static string? DefaultTopic;
        public static string? DefaultPayload;
        public static string? DefaultContentType;
        public static int? DefaultFormatIndicator;
        public static int? DefaultCorrelationIndex;
        public static int? DefaultQos;
        public static TestCaseDuration? DefaultMessageExpiry;
        public static string? DefaultResponseTopic;
        public static int? DefaultSourceIndex;

        public string? Topic { get; set; } = DefaultTopic;

        public string? Payload { get; set; } = DefaultPayload;

        public bool BypassSerialization { get; set; }

        public string? ContentType { get; set; } = DefaultContentType;

        public int? FormatIndicator { get; set; } = DefaultFormatIndicator;

        public Dictionary<string, string> Metadata { get; set; } = new();

        public int? CorrelationIndex { get; set; } = DefaultCorrelationIndex;

        public string? CorrelationId { get; set; }

        public int? Qos { get; set; } = DefaultQos;

        public TestCaseDuration? MessageExpiry { get; set; } = DefaultMessageExpiry;

        public string? ResponseTopic { get; set; } = DefaultResponseTopic;

        public int? SourceIndex { get; set; } = DefaultSourceIndex;

        public int? PacketIndex { get; set; }
    }
}
